js/util.js

```diff
@@ -5,10 +5,48 @@ app.util = (function() {
 	'use strict';
 
 	return {
+
 		extend: function() {
+
 			var args = Array.prototype.slice.call(arguments);
 			return _.extend.apply(_, [{}].concat(args));
+		},
+
+		generateRandomString: function(length, charset) {
+
+			var randString = '';
+
+			charset = charset || 'abcdefghijklmnopqrstuvqxyzABCDEFGHIJKLMNOPQRSTUVQXYZ1234567890';
+
+			var randomValues = app.util.getRandomValues(length);
+
+			for (var index = 0; index < length; index++) {
+				randString += charset[randomValues[index] % charset.length];
+			}
+
+			return randString;
+		},
+
+		getRandomValues: function(length) {
+
+			var randomValues;
+
+			// This should work for the latest browsers:
+			if (window.crypto && window.crypto.getRandomValues) {
+				randomValues = new Uint32Array(length);
+				window.crypto.getRandomValues(randomValues);
+			// Fallback for IE:
```

Comment:
Do you really want to support IE? Is it not meant to be wrapped and ran on some Chrome-like browser with Cordova?

Reply:
Good point... maybe it should fail spectacularly in the case of IE.